core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -2818,12 +2825,28 @@ object SparkContext extends Logging {
         // multiple executor resources.
         val resourceNumSlots = Math.floor(execAmount * taskReq.numParts / taskReq.amount).toInt
         if (resourceNumSlots < numSlots) {
+          if (shouldCheckExecCores) {
+            throw new IllegalArgumentException("The number of slots on an executor has to be " +
```

Comment:
only throw this exception when dynamic allocation is enabled?

Reply:
I think its safer to always require it just in case there are other places in the code that use cores and task cpus to determine slots.  I know in doing the stage level sched work there were a bunch of places that did this but I would have to go back thru to see if they were only during dynamic allocation. 
Actually one example of this is #27126

Reply:
sounds good